### PR TITLE
Add Public Holidays logic

### PR DESCRIPTION
**Pull request by ShekharGupta1988**
I am looking to move this config out of code so that it can be updated. Few things needs to be ironed out here.

### CODE REVIEW

File: src/app/liveChatSettings.ts [hunk 1]
```diff
@@ -32,4 +32,37 @@ export class LiveChatSettings {
     public static BuisnessEndDay: Number = 5;   // Friday
     public static BusinessStartHourPST: Number = 7;   // 7 AM PST
     public static BusinessEndHourPST: Number = 17;     // 5 PM PST
+
+    public static PublicHolidays: any = [
```
Comment (stevenernst131) [inline src/app/liveChatSettings.ts]: Might be worth looking into this: https://www.npmjs.com/package/moment-holiday

Comment (stevenernst131) [reply src/app/liveChatSettings.ts]: there is a moment.isHoliday() function it looks like

Comment (ShekharGupta1988) [reply src/app/liveChatSettings.ts]: oh nice.. We still need to pass a list of holidays. I'll explore that. looks to be a better way

File: src/app/shared/services/livechat.service.ts [hunk 2]
```diff
@@ -206,8 +206,23 @@ export class LiveChatService {
...
         return isApplicable;
     }
+
+    private isPublicHoliday(currentDate): boolean {
+
+        for (var iter = 0; iter < LiveChatSettings.PublicHolidays.length; iter++) {
+
+            var element = LiveChatSettings.PublicHolidays[iter];
+
+            if (element.date == currentDate.date() && ((element.month - 1) == currentDate.month()) && element.year == currentDate.year()) {
```
Comment (stevenernst131) [inline src/app/shared/services/livechat.service.ts]: There might be some moment utilities to do some of these comparisons

File: src/app/liveChatSettings.ts (hunk 1 above)
Comment (ShekharGupta1988) [reply src/app/liveChatSettings.ts]: There is little more exploration needed here. For eg- 'Good Friday' is not a holiday in Microsoft but this moment-holiday returns true. So chat will not show on that day.

I'll explore it little bit more and see how to best use this. If I just use it default, there is an risk of not showing chat on few days.

Comment (stevenernst131) [reply src/app/liveChatSettings.ts]: Makes sense, did you get the ones you added from the official Microsoft holiday list?

Comment (ShekharGupta1988) [reply src/app/liveChatSettings.ts]: Right now, I only added for remainder of this year.

**Pull request merged**